api/v1alpha1/cachefsruntime_types.go

```diff
@@ -0,0 +1,203 @@
+/*
+Copyright 2024 The Fluid Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	CacheFSRuntimeKind = "CacheFSRuntime"
+)
+
+// CacheFSRuntimeSpec defines the desired state of CacheFSRuntime
+type CacheFSRuntimeSpec struct {
+	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
+	// Important: Run "make" to regenerate code after modifying this file
+
+	// The version information that instructs fluid to orchestrate a particular version of CacheFS.
+	CacheFSVersion VersionSpec `json:"cachefsVersion,omitempty"`
+
+	// The component spec of CacheFS worker
+	Worker CacheFSCompSpec `json:"worker,omitempty"`
+
+	// Desired state for CacheFS Fuse
+	Fuse CacheFSFuseSpec `json:"fuse,omitempty"`
+
+	// Tiered storage used by CacheFS
+	TieredStore TieredStore `json:"tieredstore,omitempty"`
+
+	// Dataset used as CacheFS source
+	ImportDataset string `json:"importDataset,omitempty"`
+
+	// Configs of CacheFS, options for cachefs format sub-command, e.g. ["storage=oss", "capacity=100"]
+	Configs *[]string `json:"configs,omitempty"`
```

Comment:
How about using map instead of slice?

Reply:
Sure, I can change it to map.

I'm just curious that is there any downside of slice?